src/platforms/esp32/components/avm_builtins/network_driver.c

```diff
@@ -335,6 +335,7 @@ static void event_handler(void *arg, esp_event_base_t event_base, int32_t event_
             case IP_EVENT_STA_GOT_IP: {
                 ip_event_got_ip_t *event = (ip_event_got_ip_t *) event_data;
                 ESP_LOGI(TAG, "IP_EVENT_STA_GOT_IP: %s", inet_ntoa(event->ip_info.ip));
+                esp_sntp_restart();
```

Comment:
Should there be a check to make sure sntp was running to begin with?

Reply:
not needed as the underlying function does it:
https://github.com/espressif/esp-idf/blob/707d097b01756687cca18be855a2675d150247ae/components/lwip/apps/sntp/sntp.c#L123

```
bool sntp_restart(void)
{
    if (sntp_enabled()) {
        SNTP_ERROR(tcpip_callback(sntp_do_restart, NULL), "sntp_restart: tcpip_callback() failed");
        return true;
    }
    return false;
```
}